### PR TITLE
Adsk Contrib - Add dynamic properties and exposure contrast.

### PR DESCRIPTION
**Pull request by BernardLefebvre**
Exposure contrast has 3 properties that can be dynamic. When a property is dynamic it can be change after the processor has been created and optimized.
If 2 ops do have the same dynamic property, both will be controlled together.
Rework GPU unit test infrastructure to allow the same shader to be used several time (so that dynamic properties can be tested).
See issue #741

### CODE REVIEW

File: include/OpenColorIO/OpenColorTransforms.h [hunk 1]
```diff
@@ -388,6 +388,30 @@ OCIO_NAMESPACE_ENTER
...
+    //             (after finalization).  For example, to modify the exposure
+    //             in a viewport.
+    class OCIOEXPORT DynamicProperty
+    {
+    public:
+
+        //!cpp:function:: 
+        virtual DynamicPropertyValueType getValueType() const = 0;
+        //!cpp:function:: 
+        virtual double getDoubleValue() const = 0;
```
[the remark this answers is not in the document]
Comment (BernardLefebvre) [reply include/OpenColorIO/OpenColorTransforms.h]: I was planning to add bool getBoolValue() for the boolean dynamic property.
For the setValue, the parameter makes it clear what type of value is set and it looks simpler to not add the type in the function name.

File: include/OpenColorIO/OpenColorTransforms.h [hunk 2]
```diff
@@ -498,7 +522,113 @@ OCIO_NAMESPACE_ENTER
...
+        //                units of stops (regardless of style), for example,
+        //                a value of -1 would be equivalent to reducing the
+        //                lighting by one half.
+        double getExposure() const;
+        //!cpp:function::
+        void setExposure(double exposure);
+        //!cpp:function::
+        bool isExposureDynamic() const;
+        //!cpp:function::
+        void makeExposureDynamic();
```
[the remark this answers is not in the document]
Comment (BernardLefebvre) [reply include/OpenColorIO/OpenColorTransforms.h]: I did not see any need to allow used to make a property dynamic and then change it back to not being dynamic. If that's needed at some pont then the function can be added.
Transforms are created and configured and then used to create a processor.
With a parmeter should the function be renamed to setExposureDynamic(bool)?

[the remark this answers is not in the document]
Comment (michdolan) [reply include/OpenColorIO/OpenColorTransforms.h]: Yeah, that makes sense. Agreed on the name change if that is ever needed.

File: include/OpenColorIO/OpenColorTransforms.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (michdolan) [reply include/OpenColorIO/OpenColorTransforms.h]: Right, so you'll just overload setValue. Makes sense.

**Comment by doug-walker** [comment]
Since this PR has been open for almost two weeks, we plan to merge at the end of this week if there are no more pending comments.  Thx

**Pull request merged**